From ec8ce1f02ed81e07b7ad7c79d94e1401ca549354 Mon Sep 17 00:00:00 2001
From: David Reaver <me@davidreaver.com>
Date: Mon, 3 Feb 2025 13:13:09 -0800
Subject: [RFC PATCH 1/6] debugfs: Add temporary "#define debugfs_node dentry"
 directives
To: Greg Kroah-Hartman <gregkh@linuxfoundation.org>,
    Rafael J. Wysocki <rafael@kernel.org>,
    Danilo Krummrich <dakr@kernel.org>
Cc: Steven Rostedt <rostedt@goodmis.org>,
    Christian Brauner <brauner@kernel.org>,
    Alexander Viro <viro@zeniv.linux.org.uk>,
    Tejun Heo <tj@kernel.org>,
    linux-fsdevel@vger.kernel.org,
    cocci@inria.fr,
    linux-kernel@vger.kernel.org,
    David Reaver <me@davidreaver.com>

Add some temporary "#define debugfs_node dentry" directives to facilitate
migrating debugfs APIs from dentry to a new opaque descriptor,
debugfs_node. Subsequent commits that replace dentry with debugfs_node rely
on these #defines until the final commit in this series removes them.

This is also added to dcache.h, right below struct dentry, so it is
available where dentry is transitively included.

Signed-off-by: David Reaver <me@davidreaver.com>
---
 include/linux/dcache.h  | 2 ++
 include/linux/debugfs.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/linux/dcache.h b/include/linux/dcache.h
index 4afb60365675..4b0c11cd3d50 100644
--- a/include/linux/dcache.h
+++ b/include/linux/dcache.h
@@ -127,6 +127,8 @@ struct dentry {
 	} d_u;
 };
 
+#define debugfs_node dentry
+
 /*
  * dentry->d_lock spinlock nesting subclasses:
  *
diff --git a/include/linux/debugfs.h b/include/linux/debugfs.h
index fa2568b4380d..e6ee571e8c36 100644
--- a/include/linux/debugfs.h
+++ b/include/linux/debugfs.h
@@ -21,6 +21,8 @@
 struct device;
 struct file_operations;
 
+#define debugfs_node dentry
+
 struct debugfs_blob_wrapper {
 	void *data;
 	unsigned long size;
